GitUI/CommandsDialogs/BrowseDialog/GitStatusMonitor.cs

```diff
@@ -199,18 +200,23 @@ private GitStatusMonitorState CurrentStatus
 
                     case GitStatusMonitorState.Running:
                         {
-                            _timerRefresh.Start();
-                            if (Directory.Exists(_workTreeWatcher.Path))
-                            {
-                                _workTreeWatcher.EnableRaisingEvents = true;
-                            }
+                            _workTreeWatcher.EnableRaisingEvents = Directory.Exists(_workTreeWatcher.Path);
 
-                            if (Directory.Exists(_gitDirWatcher.Path))
+                            _gitDirWatcher.EnableRaisingEvents = Directory.Exists(_gitDirWatcher.Path) &&
+                                    !_gitDirWatcher.Path.StartsWith(_workTreeWatcher.Path);
+
+                            lock (_statusSequence)
                             {
-                                _gitDirWatcher.EnableRaisingEvents = !_gitDirWatcher.Path.StartsWith(_workTreeWatcher.Path);
+                                // An interactive update will be requested separately
+                                _nextUpdateTime =
+                                    _nextEarliestTime =
+                                    Environment.TickCount + FileChangedUpdateDelay;
```

Comment:
@mstv I like your version even less than @gerhard's.

Reply:
I agree regarding multiple assignments. Though it will need very good arguments to convince me that operators shouldn't be put in front of the continuation line.

Reply:
I can duplicate the assignments too, but changing to having the operator first right now.

Reply:
I will change this according to mstv's proposal rather than duplicating the assignment